solutions/cf-20C.mdx

```diff
@@ -0,0 +1,74 @@
+---
+id: cf-20C
+source: CF
+title: Dijkstra?
+author: Jesse Choe
+---
+
+We can implement Dijkstra's Shortest Path algorithm to find the shortest distances between node $1$ and node $n$. To find the path that yields the shortest path between nodes $1$ and $n$, we should backtrack from node $n$ and determine if a given node should be in the path using the shortest distance array. However, when the shortest distance from node 1 to node $n$ is $\infty$, then we should print $-1$. 
+
+**Time Complexity:** $\mathcal{O}(N\log M)$
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+using ll = long long;
+using vi = vector<int>;
+using pi = pair<ll, ll>;
+
+#define f first
+#define s second
+#define all(x) begin(x), end(x)
+
+int main() {
+	priority_queue<pi, vector<pi>, greater<pi>> pq;
+	int n, m; cin >> n >> m;
+	for(int i = 0; i < m; i++){
+		int a, b, w; cin >> a >> b >> w;
+		a--, b--; adj[a].push_back({w, b}), adj[b].push_back({w, a});
+	}
+	vector<ll> dist(n, INF); dist[0] = 0;
```

Comment:
did you test the code?? what's `INF`?

Reply:
shit mb forgot to add this after removing my template.